docs/development/design/stack.md

```diff
@@ -0,0 +1,60 @@
+# Nexodus Stack
+
+## Summary
+
+This proposal documents the decisions made around the Nexodus software stack.
+
+## Proposal
+
+### Kubernetes Only
+
+#### Background
+
+Early on, we used Docker Compose for orchestration.
+As the stack and complexity grew, we decided to exclusively support Kubernetes, requiring local development to use KIND.
+
+To support use of KIND for local development, and OpenShift for production deployment we settled on using Kustomize.
+
+#### Decisions
+
+- We only support deployment on K8s
+- Kustomize is our deployment configuration tool
+- All manifests in `deploy/base` MUST be useable in any K8s distribution.
+- We allow operators to be used, provided they may also be installed on any K8s distribution.
+
+### Microservices First
+
+#### Background
+
+An early agreed architectural principal of the Nexodus stack was to embrace microservices. This would allow us to iterate on components seperately, give us maximum re-use if the project failed, and most importantly, allow us to scale each component seperately.
```

Comment:
While there is nothing wrong microservices, I think it's important to note when they are most useful, and thats when you can delegate the operation of each service to a different group and thus decouple and scale the organization easier.  If 1 group has to run and maintain multiple services, sometimes a monolith makes more sense and has lower overheads.  So I think we need to be careful what we choose to make a microservice.